protocol/messenger.go

```diff
@@ -2499,7 +2507,7 @@ func (m *Messenger) getLastClockWithRelatedChat() (uint64, *Chat) {
 }
 
 // SendPairInstallation sends a pair installation message
-func (m *Messenger) SendPairInstallation(ctx context.Context) (*MessengerResponse, error) {
+func (m *Messenger) SendPairInstallation(ctx context.Context, rawMessageHandler RawMessageHandler) (*MessengerResponse, error) {
```

Comment:
This isn't something that should to be changed in this PR, but perhaps it could be in the scope for discussion.

First point I want to make is that I really like the way you capture the messages from `Messenger`. It is a nice solution, it reaches into `Messenger` grabs message data we want and then the `pairing` processes can take over from this point.

One thing I notice is that a lot of the syncing methods now have a nilable parameter, that seems to be nil in the majority of cases. Perhaps we could simplify the method logic and update `Messenger`'s state with a "RawMessageHandler" register.

The consuming code would look like something below:

```go
m.SetRawMessageHandler(rawMessageHandler)
m.SyncDevices()
m.SendPairInstallation(context.TODO)
m.ResetRawMessageHandler() // Defaults back to `dispatchPairInstallationMessage()`
```

Nothing to change now. But implementing this change would simplify the calling of all the functions that have an optional `RawMessageHandler `, future developers don't need to know what a `RawMessageHandler` is unless they want to change "broadcasting" behaviour.